dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java

```diff
@@ -423,11 +423,12 @@ private void saveWebAsset(Map<String, Object> contentletFormData,
 			}
 		 }
 
-		if (categoriesList != null && categoriesList.size() > 0) {
-			for (final Iterator iterator = categoriesList.iterator(); iterator.hasNext();) {
-
-				final String tmpString = (String) iterator.next();
-				categories.add(catAPI.find(tmpString, user, false));
+		if (UtilMethods.isSet(categoriesList)) {
+			for (final Iterator<String> iterator = categoriesList.iterator(); iterator.hasNext();) {

```

Comment:
I would replace the iterator with a normal `for` over `categoriesList` 
```for (final String categoryId : categoriesList)```

Reply:
Yes, I thought about it too. But didn't add many changes as this is very old code that will be moved over to a REST Endpoint, hopefully soon.